components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/xpath/VaultLookupFunction.java

```diff
@@ -96,4 +97,18 @@ private void traceOrDebug(boolean traceOn, String msg) {
 		}
 	}
 
+	private SecretSrcData getSecretSourceInfo (List args) {
+		if (args.size() == 3) {
```

Comment:
I feel like you need to just call new SecretSrcData(args) here and all this validations done inside this method have to be moved to the constructor of SecretSrcData. WDYT?